cli/runtime/config/config_factory.go

```diff
@@ -8,22 +8,161 @@ import (
 	"os"
 
 	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/config/collectionutils"
+
 	"gopkg.in/yaml.v3"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/config/nodeutils"
 )
 
-type CfgOptions struct {
-	CfgPath string // file path to the config file
+// getMultiConfig retrieves combined config.yaml and config-ng.yaml
+func getMultiConfig() (*yaml.Node, error) {
+	cfgNode, err := getClientConfig()
+	if err != nil {
+		return cfgNode, err
+	}
+
+	cfgNextGenNode, err := getClientConfigNextGenNode()
+	if err != nil {
+		return cfgNextGenNode, err
+	}
+
+	return makeMultiFileCfg(cfgNode, cfgNextGenNode)
+}
+
+// getMultiConfigNoLock retrieves combined config.yaml and config-ng.yaml
+func getMultiConfigNoLock() (*yaml.Node, error) {
+	cfgNode, err := getClientConfigNoLock()
+	if err != nil {
+		return cfgNode, err
+	}
+
+	cfgNextGenNode, err := getClientConfigNextGenNodeNoLock()
+	if err != nil {
+		return cfgNextGenNode, err
+	}
+
+	return makeMultiFileCfg(cfgNode, cfgNextGenNode)
+}
+
+// Construct multi file node from config.yaml and config-ng.yaml nodes
+func makeMultiFileCfg(cfgNode, cfgNextGenNode *yaml.Node) (*yaml.Node, error) {
+	// config nodes that goes to config.yaml
+	cfgItems := []string{
+		KeyAPIVersion,
```

Comment:
nit: perhaps the the list itself should be a constant since it is used in more than one place?

Reply:
renaming